ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModel.kt

```diff
@@ -41,7 +44,10 @@ class PreviewImageViewModel : ViewModel() {
             else -> ImageInHighResLoadSuccessUiState
         }
 
-        if (newState == ImageInHighResLoadSuccessUiState && currentState != ImageInHighResLoadSuccessUiState) {
+        if (newState == ImageInHighResLoadSuccessUiState &&
+                currentState != ImageInHighResLoadSuccessUiState &&
+                loadIntoFile.value !is ImageLoadToFileSuccessState
```

Comment:
I feel like this condition is starting to be difficult to reason about. How would you feel about something like this?

```
val highResImageJustLoadedIntoView = newState != currentState && newState == ImageInHighResLoadSuccessUiState
val imageNotLoadedIntoFile  = loadIntoFile.value !is ImageLoadToFileSuccessState
if (highResImageJustLoadedIntoView && imageNotLoadedIntoFile) {
...
```

Reply:
Agree. Changed here dd96921